plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/BuildPath.java

```diff
@@ -104,12 +110,24 @@ public static void addMavenLibraries(IProject project, List<Library> libraries,
       }
       subMonitor.worked(1);
     }
-    javaProject.setRawClasspath(newRawClasspath.toArray(new IClasspathEntry[0]), subMonitor);
+    
+    if (addToClasspath) {
+      javaProject.setRawClasspath(newRawClasspath.toArray(new IClasspathEntry[0]), subMonitor);
+    }
     
     runContainerResolverJob(javaProject);
```

Comment:
This seems a bit tricky.

It's not entirely clear what this `runContainerResolverJob()` does, but it looks like it should do something about newly added libraries. However, when we are adding libraries through the _Build Path_ menu, this job may run and complete (i.e., having no effect) before libraries are actually added later by the `getNewContainers()` interface contract.